presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -84,15 +86,17 @@
     private final AtomicLong finishUserNanos = new AtomicLong();
 
     private final AtomicLong memoryReservation = new AtomicLong();
-    private final AtomicLong revocableMemoryReservation = new AtomicLong();
+    @GuardedBy("this")
```

Comment:
Generally it seems to me that it would look better split. But I would not invest too much in that. Let's see what FB says eventually. Just leave the comment about alternative approaches in the code.